cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -547,43 +643,115 @@ fn insert_common_epilogues(
     stack_size: i64,
     reg_type: ir::types::Type,
     csrs: &RegisterSet,
+    isa: &dyn TargetIsa,
+    cfa_state: &CFAState,
 ) {
     while let Some(ebb) = pos.next_ebb() {
         pos.goto_last_inst(ebb);
         if let Some(inst) = pos.current_inst() {
             if pos.func.dfg[inst].opcode().is_return() {
-                insert_common_epilogue(inst, stack_size, pos, reg_type, csrs);
+                // figure out if we need to insert end-of-function-aware frame layout information
+                let following_inst = pos.next_ebb().and_then(|next_ebb| {
+                    pos.goto_first_inst(next_ebb);
+                    pos.current_inst()
```

Comment:
You can use [`pos.func.layout.first_inst`](https://docs.rs/cranelift-codegen/0.43.1/cranelift_codegen/ir/layout/struct.Layout.html#method.first_inst) instead of `goto_first_inst`, `current_inst` and `goto_last_inst`.

Reply:
ah so `following_inst` would be
```
let following_inst = pos.func.layout().next_ebb(ebb)
  .and_then(|next_ebb| pos.func.layout().first_inst(next_ebb))
```
? that cleans this up nicely.